s3_log_shipper/parsers.py

```diff
@@ -0,0 +1,123 @@
+import json
+import os
+from dataclasses import dataclass
+from datetime import datetime
+from pathlib import Path
+from typing import Optional, List, Tuple
+
+from pygrok import Grok
+
+import s3_log_shipper
+
+
+@dataclass
+class Parser:
+    """
+    Encapsulates the grok patterns to identify a specific log file and parse the logs within it.
+    """
+
+    type: str
+    log_grok: Grok
+    path_groks: List[Grok]
+    strptime_pattern: str
+
+    def match_path(self, log_path) -> Optional[dict]:
```

Comment:
Since you've alread spoiled the reviewer by annotating types, is it possible to annotate the type of `log_path` please